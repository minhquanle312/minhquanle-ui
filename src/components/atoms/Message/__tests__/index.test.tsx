// Libraries
import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';

// Components
import { message } from '../index';
import { Button } from '../../index';

const MessageTest = () => {
  const [messageApi, contextHolder] = message.useMessage();

  const info = () => {
    messageApi.info('Hello, Ant Design!');
  };

  return (
    <>
      {contextHolder}
      <Button type="primary" onClick={info}>
        Display normal message
      </Button>
    </>
  );
};

describe('Message', () => {
  it('should match snapshot', () => {
    const loadingIndicator = render(<MessageTest />);
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
  it('should display normal message when button is clicked', async () => {
    const { getByText, getByRole } = render(<MessageTest />);

    fireEvent.click(getByRole('button'));

    await waitFor(() => {
      expect(getByText('Hello, Ant Design!')).toBeInTheDocument();
    });
  });
});
