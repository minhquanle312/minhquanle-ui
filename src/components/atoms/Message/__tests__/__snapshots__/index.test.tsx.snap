// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should match snapshot 1`] = `
<button
  class="ant-btn css-dev-only-do-not-override-htwhyh ant-btn-primary"
  type="button"
>
  <span>
    Display normal message
  </span>
</button>
`;
